source/slang/slang-parameter-binding.cpp

```diff
@@ -531,49 +537,67 @@ LayoutSemanticInfo ExtractLayoutSemanticInfo(
         index = index * 10 + (c - '0');
     }
 
-
     UInt space = 0;
-    if( auto registerSemantic = as<HLSLRegisterSemantic>(semantic) )
+    if(spaceName.getLength() != 0)
     {
-        auto const& spaceName = registerSemantic->spaceName.getContent();
-        if(spaceName.getLength() != 0)
-        {
-            UnownedStringSlice spaceSpelling;
-            UnownedStringSlice spaceDigits;
-            splitNameAndIndex(spaceName, spaceSpelling, spaceDigits);
+        UnownedStringSlice spaceSpelling;
+        UnownedStringSlice spaceDigits;
+        splitNameAndIndex(spaceName, spaceSpelling, spaceDigits);
 
-            if( kind == LayoutResourceKind::RegisterSpace )
-            {
-                getSink(context)->diagnose(registerSemantic->spaceName, Diagnostics::unexpectedSpecifierAfterSpace, spaceName);
-            }
-            else if( spaceSpelling != UnownedTerminatedStringSlice("space") )
-            {
-                getSink(context)->diagnose(registerSemantic->spaceName, Diagnostics::expectedSpace, spaceSpelling);
-            }
-            else if( spaceDigits.getLength() == 0 )
-            {
-                getSink(context)->diagnose(registerSemantic->spaceName, Diagnostics::expectedSpaceIndex);
-            }
-            else
+        if( kind == LayoutResourceKind::RegisterSpace )
+        {
+            sink->diagnose(spaceLoc, Diagnostics::unexpectedSpecifierAfterSpace, spaceName);
+        }
+        else if( spaceSpelling != UnownedTerminatedStringSlice("space") )
```

Comment:
Could use UnownedStringSlice::fromLiteral to remove the strlen (although compiler may do so anyway).

Reply:
Fair point, although that is pre-existing code that I just moved around for a refactor, so I'm disinclined to change it during an unrelated PR.